dnsaas/admin.py

```diff
@@ -0,0 +1,427 @@
+    list_per_page = 250
+    save_on_top = True
+    search_fields = ('name', 'content',)
+    readonly_fields = ('change_date', 'ordername', 'created', 'modified')
+    related_search_fields = {
+        'domain': ('name',),
+    }
+    fieldsets = (
+        (None, {
+            'fields': (
+                'owner',
+                'domain',
+                ('type', 'name', 'content',),
+                'auth',
+            ),
+        }),
+        ('Advanced options', {
+            'classes': ('collapse',),
+            'fields': ('prio', 'ttl', 'ordername', 'change_date',)
+        }),
+        (None, {'fields': ('created', 'modified')})
+    )
+    formfield_overrides = {
+        models.NullBooleanField: {
+            'widget': NullBooleanRadioSelect(
+                attrs={'class': 'radiolist inline'}
+            ),
+        },
+    }
+    # CopyingAdmin
+    FromModel = Domain
+    CopyFieldsModel = Domain
+    from_field = 'domain'
+    field_prefix = 'record_'
+
+    def get_form(self, request, *args, **kwargs):
+        form = super().get_form(request, *args, **kwargs)
+        form.user = request.user
+        return form
+
+RECORD_LIST_FIELDS = (
+    'name',
+    'type',
+    'content',
+    'ttl',
+    'prio',
+)
+class RecordTemplateAdmin(ForeignKeyAutocompleteAdmin):
+    form = RecordAdminForm
+    list_display = RECORD_LIST_FIELDS
+
+
+class SuperMasterAdmin(admin.ModelAdmin):
+    list_display = ('ip', 'nameserver', 'account',)
+    list_filter = ('ip', 'account', 'created', 'modified')
+    search_fields = ('ip', 'nameserver',)
+    readonly_fields = ('created', 'modified')
+
+
+class DomainMetadataAdmin(ForeignKeyAutocompleteAdmin):
+    list_display = ('domain', 'kind', 'content',)
+    list_filter = ('kind', 'domain', 'created', 'modified')
+    list_per_page = 250
+    list_filter = ('created', 'modified')
+    readonly_fields = ('created', 'modified')
+    related_search_fields = {
+        'domain': ('name',),
+    }
+    save_on_top = True
+    search_fields = ('content',)
+
+class CryptoKeyAdmin(ForeignKeyAutocompleteAdmin):
+    list_display = ('domain', 'flags', 'active', 'content',)
+    list_filter = ('active', 'domain', 'created', 'modified')
+    list_per_page = 250
+    readonly_fields = ('created', 'modified')
+    related_search_fields = {
+        'domain': ('name',),
+    }
+    save_on_top = True
+    search_fields = ('content',)
+    formfield_overrides = {
+        models.NullBooleanField: {
+            'widget': NullBooleanRadioSelect(
+                attrs={'class': 'radiolist inline'}
+            ),
+        },
+    }
+
+class RecordTemplateInline(admin.StackedInline):
+    model = RecordTemplate
+    extra = 1
+class DomainTemplateAdmin(ForeignKeyAutocompleteAdmin):
+    inlines = [RecordTemplateInline]
+    list_display = ['name', 'add_domain_link', 'is_public_domain']
+
+
+
+
+
+#TODO:: restrict all only for superuser
```

Comment:
consider adding (simple) admin for record requests (only for preview, no possiblity to add new one)